app/src/main/java/com/gogiyumandroid/ui/home/HomeFragment.kt

```diff
@@ -1,31 +1,85 @@
+@file:Suppress("DEPRECATION")
+
 package com.gogiyumandroid.ui.home
 
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import android.widget.TextView
+import android.widget.ImageView
+import android.widget.TableLayout
+import android.widget.TableRow
 import androidx.fragment.app.Fragment
-import androidx.lifecycle.Observer
 import androidx.lifecycle.ViewModelProviders
+import com.bumptech.glide.Glide
 import com.gogiyumandroid.R
+import kotlinx.android.synthetic.main.custom_table_row.view.*
+import okhttp3.*
+import org.json.JSONArray
+import java.io.IOException
+
 
 class HomeFragment : Fragment() {
 
     private lateinit var homeViewModel: HomeViewModel
+    private val myGogiyumAddress = "https://gogiyum.com/food.json"
 
     override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
+            inflater: LayoutInflater,
+            container: ViewGroup?,
+            savedInstanceState: Bundle?
     ): View? {
         homeViewModel =
-            ViewModelProviders.of(this).get(HomeViewModel::class.java)
+                ViewModelProviders.of(this).get(HomeViewModel::class.java)
+
+
+
         val root = inflater.inflate(R.layout.fragment_home, container, false)
-//        val textView: TextView = root.findViewById(R.id.text_home)
-        homeViewModel.text.observe(viewLifecycleOwner, Observer {
-//            textView.text = it
-        })
+        val tableLayout = root.findViewById<TableLayout>(R.id.TableLayout1)
+
+//        tableRow = root.findViewById<TableRow>(R.id.table_row1)
+        readData(myGogiyumAddress, tableLayout)
+
         return root
     }
+
+//    synchronous
+    private fun readData(url: String, tableLayout: TableLayout) {
+        val request = Request.Builder().url(url).build()
+        val client = OkHttpClient()
+
+        client.newCall(request).enqueue(object : Callback {
+            override fun onResponse(call: Call, response: Response) {
+                val body = response.body()?.string()
+                val jsonArray = JSONArray(body)
+
+                activity?.runOnUiThread {
+                    handleTableRow(jsonArray, tableLayout)
+                }
+            }
+            override fun onFailure(call: Call, e: IOException) {
+                println("fail")
+            }
+        })
+    }
+
+    fun handleTableRow(jsonArray: JSONArray, tableLayout: TableLayout ){
+        for (i in 0 until jsonArray.length()) {
+            val jsonObject = jsonArray.getJSONObject(i)
+                val tableRows = TableRow(activity)
+                layoutInflater.inflate(R.layout.custom_table_row, tableRows, true)
+                tableRows.textView4.text = jsonObject.getString("name")
```

Comment:
textView4 is not good naming. 
Can you change better name? 
i.e) foodImageName

Reply:
I changed it